PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,326 @@
+            break
+    if not submit:
+        logging.info("Advanced Search button not found！")
+        return
+    submit.click()
+
+
+def login_myportal(driver):
+    """
+    Open myportal website and login.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    # Open the browser (The true url is：https://myportal.fhda.edu/）
+    driver.get("https://myportal.fhda.edu/")
+    # input username
+    username = int(input("Please enter your user name:  "))
+    driver.find_element_by_id("j_username").send_keys(username)
+    # input password
+    password = input("Please enter your password:   ")
+    driver.find_element_by_id("j_password").send_keys(password)
+    driver.find_element_by_id(
+        "btn-eventId-proceed").click()  # Wait for the response from the next page and make sure the page is loaded!
+    logging.info("Log in finished.")
+
+
+# Open course search page
+def openSearchPage(driver):
+    """
+    Click Appls->Look Up Classes-> open search page.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    # Find the Apps menu and click
+    findAppsMenu(driver)
+    # Find in App List（Look Up Classes）button
+    classes = lookUpClasses(driver)
+    mainWindowName = driver.window_handles[0]  # Main window name
+    classes.click()
+    time.sleep(2)  # Make sure the new window is open
+    # Switch to the newly opened window, because this window is the course search page, you must switch the program pointer to this window to operate this window
+    windowNames = driver.window_handles  # Get all window names
+    for name in windowNames:
+        if mainWindowName == name:
+            continue  # Do not switch to the main window because the search page opens from the main window
+        driver.switch_to_window(name)
+        break  # Because there are only two windows: the main window and the open window, switching directly to the end is over
+    # Waiting for elements in the page to appear, indicating that the page has finished loading
+    waitUtilPageLoaded(driver)
+
+
+def findAppsMenu(driver):
+    """
+    Find Apps menu.
+
+    Parameters: webdriver
+    Returns: None
+    """
+    menus = driver.find_elements_by_class_name("list-group-item")
+    '''if not menus:
+        logging.info("No left menu found！")
+        return'''
+    appMenu = []
+    for menu in menus:
+        txt = menu.text
+        if "apps".lower() == txt.lower():  # No left menu found
+            appMenu = menu
+            break
+    if not appMenu:
+        logging.info("Apps menu not found in left menu！")
+        return
+    time.sleep(2)  # Wait for the next page to come over
+    appMenu.click()  # Open the menu
+
+
+def lookUpClasses(driver):
+    """
+    Find app list.
+
+    Parameters: webdriver
+    Returns: classes
+    Return type: list
+    """
+    myappsclasses = driver.find_elements_by_class_name("myapps-item")
+    '''if not myappsclasses:
+        logging.info("No app list found！")
+        return'''
```

Comment:
You may want to remove this (if this was for debugging)

Reply:
Actually I wanted to save these lines, but due to the 250 lines control of total codes, I have to remove something.